pinc/Project.inc

```diff
@@ -1317,3 +1311,47 @@ function get_page_image_param($arr, $key, $allownull = false)
     }
     return $page_name;
 }
+
+
+// XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+// Image source functions
+
+function load_image_sources()
+{
+    static $image_sources = [];
+    if ($image_sources) {
+        return $image_sources;
+    }
+
+    $sql = "
+        SELECT *
+        FROM image_sources
+        ORDER BY display_name
+    ";
+    $result = DPDatabase::query($sql);
+    while ($row = mysqli_fetch_assoc($result)) {
+        $image_sources[(string)$row['code_name']] = $row;
+    }
+    mysqli_free_result($result);
+    return $image_sources;
+}
+
+// Can the current user see this image source based on the source's settings
+function can_user_see_image_source($image_source)
+{
+    // info page visibility
+    //  0 = Image Source Managers and SAs
+    //  1 = also any PM
+    //  2 = also any logged-in user
+    //  3 = anyone
```

Comment:
> If we enforce this on the Show Image Sources page but not the project create/edit then PMs could conceivably select a source that they can't view which seems wrong.

I was likely following incorrect reasoning, then, because I thought the visibility had to do with archived images, not image sources. Looking back at the table fields, what I was thinking of was `ok_show_images`.

On PROD right now, the only two image sources that specify that only ISMs and squirrels can see them are also the only two that are Pending. However, on TEST, there are several that are "Pending Review" that show on the display page, because there are several that specify PMs or DP users. However, any that have the ISM/Squirrel only option are not displayed in the list. Neither form of Pending is in the select list for PMs to choose for a project. Disabled image sources are also not available in the list of options the PM has to choose from in a project. Unless I'm missing something, this all seems to be correct behavior.